### PR TITLE
A nicer error message for incorrect stage-packages

### PR DESCRIPTION
**Pull request by sergiusens**
LP: #1568131

### CODE REVIEW

File: snapcraft/internal/pluginhandler.py [hunk 1]
```diff
@@ -250,8 +250,14 @@ def _step_state_file(self, step):
...
     def _fetch_stage_packages(self):
-        if self.code.stage_packages:
+        if not self.code.stage_packages:
+            return
+
+        try:
             self.ubuntu.get(self.code.stage_packages)
+        except repo.PackageNotFoundError as e:
+            raise RuntimeError("Cannot find {!r} defined as 'stage-packages' "
+                               "for {!r}".format(e.package_name, self.name))
```
Comment (kyrofa) [inline snapcraft/internal/pluginhandler.py]: "Cannot find 'libfoo' defined as 'stage-packages' for 'bar'" doesn't read very well. What about "Error downloading stage packages for part 'bar': no such package 'libfoo'" ?

**Comment by elopio** [comment]
:+1: , but I agree with Kyle, his message is better.

File: snapcraft/internal/pluginhandler.py (hunk 1 above)
Comment (sergiusens) [inline snapcraft/internal/pluginhandler.py]: El 27/05/16 a las 11:57, Kyle Fazzari escribió:
> In snapcraft/internal/pluginhandler.py
> <https://github.com/ubuntu-core/snapcraft/pull/524#discussion_r64917665>:
> 
>>              self.ubuntu.get(self.code.stage_packages)
>> +        except repo.PackageNotFoundError as e:
>> +            raise RuntimeError("Cannot find {!r} defined as 'stage-packages' "
>> +                               "for {!r}".format(e.package_name, self.name))
> 
> "Cannot find 'libfoo' defined as 'stage-packages' for 'bar'" doesn't
> read very well. What about "Error downloading stage packages for part
> 'bar': no such package 'libfoo'" ?

sure

**Comment by sergiusens** [comment]
@kyrofa @elopio that better?

**Comment by elopio** [comment]
:100:

**Comment by kyrofa** [comment]
Beautiful! :+1:

**Pull request merged**